.github/workflows/ci.yml

```diff
@@ -360,29 +376,37 @@ jobs:
     steps:
       - name: "Checking out repository"
         uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
-        with:
-          submodules: true
+      - name: "Checking out runtime submodules"
+        run: ./build_tools/scripts/git/update_runtime_submodules.sh
       - name: Querying GPU information
         run: |
           ./build_tools/scripts/check_cuda.sh
           ./build_tools/scripts/check_vulkan.sh
-      - name: "Downloading build dir archive"
-        run: gcloud storage cp "${BUILD_DIR_GCS_ARTIFACT}" "${BUILD_DIR_ARCHIVE}"
-      - name: "Extracting build dir archive"
-        run: tar -xf "${BUILD_DIR_ARCHIVE}"
-      - name: "Testing all"
+      - name: "Downloading install dir archive"
+        run: gcloud storage cp "${INSTALL_DIR_GCS_ARTIFACT}" "${INSTALL_DIR_ARCHIVE}"
+      - name: "Extracting install directory"
+        run: tar -xf "${INSTALL_DIR_ARCHIVE}"
+      - name: "Building tests"
+        run: |
+          ./build_tools/github_actions/docker_run.sh \
+              --env IREE_CPU_DISABLE \
+              --env IREE_VULKAN_DISABLE \
+              --env IREE_CUDA_DISABLE \
+              gcr.io/iree-oss/nvidia@sha256:892fefbdf90c93b407303adadfa87f22c0f1e84b7e819e69643c78fc5927c2ba \
+              ./build_tools/pkgci/build_tests_using_package.sh ${INSTALL_DIR}
```

Comment:
between baseline and this PR, test_a100 is no longer running these tests:

* `iree/tests/e2e/stablehlo_models/mnist_train_test/mnist_train_test_cuda`
* `iree/tests/e2e/stablehlo_models/mnist_train_test/mnist_train_test_vulkan`
* `iree/tests/transform_dialect/cuda/double_mma_layout_analysis.mlir.test`
* `iree/tests/transform_dialect/cuda/mma_elemwise_layout_analysis.mlir.test`
* `iree/tests/transform_dialect/cuda/mma_reduction_layout_analysis.mlir.test`
* `iree/tests/transform_dialect/cuda/mma_using_layout_analysis.mlir.test`
* `iree/tests/transform_dialect/cuda/mma.mlir.test`

`iree/tests/transform_dialect/cuda` tests need this updated to use `IREE_HOST_BIN_DIR`:
https://github.com/openxla/iree/blob/10fd98b89616e7f1ae70b8c4036c2c715f1f5bc0/tests/transform_dialect/cuda/CMakeLists.txt#L13-L15

`mnist_train_test` I may let rot. We need a good way to plug Python tests in, and that test is .... not it right now :P